m2cgen/assemblers/boosting.py

```diff
@@ -5,36 +5,40 @@
 from m2cgen.assemblers.base import ModelAssembler
 
 
-class XGBoostModelAssembler(ModelAssembler):
+class BaseBoostAssembler(ModelAssembler):
 
-    def __init__(self, model):
-        super().__init__(model)
-        self._base_score = self.model.base_score
+    classifier_name = None
 
-        feature_names = self.model.get_booster().feature_names
-        self._feature_name_to_idx = {
-            name: idx for idx, name in enumerate(feature_names)
-        }
+    def __init__(self, model, trees, base_score=0):
+        super().__init__(model)
+        self.trees = trees
+        self._base_score = base_score
 
         self._output_size = 1
         self._is_classification = False
+
         model_class_name = type(model).__name__
-        if model_class_name == "XGBClassifier":
+        if model_class_name == self.classifier_name:
```

Comment:
We already have constructor arguments, why not pass the `classifier_name` there?

Reply:
The thing is that this parameter is not instance-specific, but rather class-specific. So I don't think it should be a constructor parameter.